src/adapters/ws/index.ts

```diff
@@ -93,15 +93,17 @@ class WebSocketsAdapter extends Adapter {
             return
           }
         }
-      }
-
-      if (servers.has(pathname)) {
-        servers.get(pathname).handleUpgrade(request, socket, head, (ws) => {
-          servers.get(pathname).emit('server:connection:open', ws, request)
-
-          ws.on('message', (payload) => {
-            const msg = this._createMessage(pathname, payload)
-            this.emit('message', msg, ws)
+        
+        if (servers.has(pathname)) {
+          servers.get(pathname).handleUpgrade(request, socket, head, (ws) => {
+            servers.get(pathname).emit('connect', ws, request)
+            
+            ws.on('message', (payload) => {
+              const msg = this._createMessage(pathname, payload)
+              this.emit('message', msg, ws)
+            })
+
+            this.emit('server:connection:open', { name: this.name(), adapter: this, connection: ws, channel: pathname, request })
           })
 
           this.emit('connect', { name: this.name(), adapter: this, connection: ws, channel: pathname })
```

Comment:
We should get rid of this `connect` event. As per documentation, `connect` is only triggered when Glee is a client but in this case it behaves as a server.